Prose/ProseLib/Sources/EditProfileFeature/Components/ThreeColumns.swift

```diff
@@ -0,0 +1,97 @@
+//
+// This file is part of prose-app-macos.
+// Copyright (c) 2022 Prose Foundation
+//
+
+import SwiftUI
+
+struct ThreeColumns<Column1: View, Column2: View, Column3: View>: View {
+  let label: String
+  let column1: () -> Column1
+  let column2: () -> Column2
+  let column3: () -> Column3
+
+  init(
+    label: String,
+    @ViewBuilder column1: @escaping () -> Column1,
+    @ViewBuilder column2: @escaping () -> Column2,
+    @ViewBuilder column3: @escaping () -> Column3
+  ) {
+    self.label = label
+    self.column1 = column1
+    self.column2 = column2
+    self.column3 = column3
+  }
+
+  init(
+    _ label: String,
+    @ViewBuilder column2: @escaping () -> Column2,
+    @ViewBuilder column3: @escaping () -> Column3
+  ) where Column1 == Text {
+    self.label = label
+    self.column1 = { Text(verbatim: label) }
+    self.column2 = column2
+    self.column3 = column3
+  }
+
+  init(
+    _ label: String,
+    @ViewBuilder column2: @escaping () -> Column2
+  ) where Column1 == Text, Column3 == EmptyView {
```

Comment:
Just tried this…

```swift
  init(
    _ label: String,
    @ViewBuilder column2: @escaping () -> Column2
  ) where Column1 == Text, Column3 == AnyView {
    self.label = label
    self.column1 = { Text(verbatim: label) }
    self.column2 = column2
    self.column3 = { AnyView(Color.clear.fixedSize()) }
  }
```

which gives the same result as using `ZStack`s below. Still not great, but feels less magic in my opinion.

Reply:
Indeed, if feels less magic, but I don't know if there's a downside to using `Color.clear` (rendering, hit testing…).

`AnyView` seems ok in this case, as it's not some content-related view.

Reply:
I just rememebered the reason why I did this:
One problem with this is that only this initializer now handles the problem. If you manually call `init(label:column1:column2:column3:)` giving an `EmptyView` (which was my case before the refactor), it breaks the layout.

Reply:
Isn't that true for every other SwiftUI container as well? Garbage in, garbage out.

Reply:
```swift
VStack(alignment: .leading) {
  Label {
    Text("test")
  } icon: {
    Text("icon")
  }
  Label {
    Text("test")
  } icon: {
    EmptyView()
  }
  Label {
    EmptyView()
  } icon: {
    Text("icon")
  }
}
.padding()
```
<img width="130" alt="Screenshot 2022-07-12 at 11 32 29" src="https://user-images.githubusercontent.com/37386490/178459267-284acc35-fb61-43b3-bb76-8c49c275b5bf.png">

Same in a `List`, looks like you're right. I'll use your proposition.